### PR TITLE
[JENKINS-35286] Upgrade to parent POM

### PR DESCRIPTION
**Pull request by kwhetstone**
[JENKINS-35286](https://issues.jenkins-ci.org/browse/JENKINS-35286)

Upgrade to new parent pom. More info [here](https://github.com/jenkinsci/plugin-pom).

- [x] Upgrade to new parent pom
- [x] Remove inherit plugin configuration
- [x] Update maven repositories

@reviewbybees

### CODE REVIEW

**Comment by kwhetstone** [comment]
@reviewbybees

File: pom.xml [hunk 1]
```diff
@@ -19,46 +19,17 @@
...
-            <goals>
-              <goal>check</goal> 
-            </goals>
-          </execution>
-        </executions>
-      </plugin>
-    </plugins>
-  </build>
+  <properties>
+    <jenkins.version>1.625.3</jenkins.version>
```
Comment (andresrc) [inline pom.xml]: What's the motivation for bumping the baseline?

**Comment by recena** [comment]
Please, add the Jira issue in the PR title and description.

File: pom.xml (hunk 1 above)
Comment (kwhetstone) [inline pom.xml]: The main motivation was getting the baseline up to a more recent LTS.  There weren't any issues with the new test harness and 1.520.  I moved it back to the old version.

File: pom.xml [hunk 2]
```diff
@@ -67,11 +38,11 @@
...
-    <pluginRepositories>
-        <pluginRepository>
-            <id>repo.jenkins-ci.org</id>
-            <url>http://repo.jenkins-ci.org/public/</url>
-        </pluginRepository>
-    </pluginRepositories>
+  <pluginRepositories>
+    <pluginRepository>
+      <id>repo.jenkins-ci.org</id>
+      <url>http://repo.jenkins-ci.org/public/</url>
```
Comment (andresrc) [inline pom.xml]: Recommended to use the same PR to change the `repositories` and `pluginRepositories` to https (see [INFRA-588](https://issues.jenkins-ci.org/browse/INFRA-588))

**Comment by kwhetstone** [comment]
Added the Jira issue id information.  I added the java.level tag as well as the changing the repositories to https.

**Comment by andresrc** [comment]
:bee: just the plugin repository missing, but that's a nit
thanks!

File: pom.xml [hunk 3]
```diff
@@ -19,59 +19,31 @@
...
-            <goals>
-              <goal>check</goal> 
-            </goals>
-          </execution>
-        </executions>
-      </plugin>
-    </plugins>
-  </build>
+  <properties>
+    <jenkins.version>1.577</jenkins.version>
```
Comment (jglick) [inline pom.xml]: :ant: I would recommend moving to 1.580.1, the next LTS, and the first to include SECURITY-144, which may necessitate a few source code changes.

File: pom.xml (hunk 2 above)
Comment (jglick) [inline pom.xml]: Yes :ant: (already done for `repositories` actually)

**Comment by jglick** [comment]
:bee:

**Comment by jglick** [comment]
BTW you have several checkboxes in the PR description which are still unchecked.

**Comment by recena** [comment]
@reviewbybees done

**Comment by recena** [comment]
@kwhetstone Technically the review is done, but I recommend to address the ants sent by @jglick and @andresrc

**Comment by kwhetstone** [comment]
The pluginRepository is there, and I bumped the version.  Anything else needed?

**Comment by jglick** [comment]
:bee:

**Comment by andresrc** [comment]
:bee:

**Comment by kwhetstone** [comment]
@reviewbybees done

**Pull request merged**